### DISCUSSION TITLE
Firefly in a reverse proxy subfolder

### DISCUSSION THREAD

**Discussion by sfuantos** [firefly-iii/firefly-iii#7702] [Firefly III]
Hello,
Has anyone made firefly fully work in a reverse proxy subfolder? What am I missing in my configuration?
I've tried different setups from issues and discussions; so far this is what works best for me but I've noticed these issues:

1. The date picker doesn't work 
2. ![Screenshot_20230630_125034](https://github.com/firefly-iii/firefly-iii/assets/22817298/a4ffc481-1abe-4702-98b0-3daf40742230)
3. When mass editing it redirects to _/transactions/mass/edit/457,456_ instead of _firefly/transactions/mass/edit/457,456_

Everything else seems to be working fine.

I'm using SWAG as the reverse proxy.

```
location /firefly/ {
    proxy_pass http://firefly:8080/;
    # Request uncompressed files from upstream server to allow manipulation
    # of response body
    proxy_set_header Accept-Encoding "";

    # Rewrite redirects, such as / to /login and /logout
    # No trailing slash - including it works for login, but (at least) not logout.
    # Also means we can't use `proxy_redirect default;`
    proxy_redirect $scheme://$host /firefly;

    # Setting proper X-Forwarded headers, in combination TRUSTED_PROXIES
    # env variable, lets Laraval create URLs based on proxied address rather
    # than server name.
    proxy_set_header X-Forwarded-For $remote_addr;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Port $server_port;
    # Real remote client ip
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host $host;

    # Add the folder name to links/asset URLs in the response body
    sub_filter $scheme://$host https://$host/firefly;
    sub_filter_once off;
}
```

**Comment by JC5** [comment]
It uses the `<base href="">` value from the source code as the source for the referer. What's that value in your case?

**Reply by sfuantos** [reply]
Inspecting the page, I see this value:

`<base href="https://mydomain.com/firefly/">`

Where mydomain.com is my actual domain.

**Reply by JC5** [reply]
Alright. Does it refer to the wrong page after you selected X transactions, and pressed the button?

**Reply by sfuantos** [reply]
Yes, it happens with both options. Edit selected individually and Edit selected in bulk.

**Reply by JC5** [reply]
This URL should display a bunch of variables: `example.com/firefly/v1/jscript/variables?ext=.js&v=6.0.16`.

Can you share the values for `mass_edit_url` and the other two? They should be near the bottom.

**Reply by sfuantos** [reply]
Here are the values:
```
var mass_edit_url   = 'https://mydomain.com/transactions/mass/edit';
var bulk_edit_url   = 'https://mydomain.com/transactions/bulk/edit';
var mass_delete_url = 'https://mydomain.com/transactions/mass/delete';
```

**Comment by JC5** [comment]
On what pages *exactly* are you trying to mass edit?

**Reply by sfuantos** [reply]
On Transactions > Expenses

**Reply by JC5** [reply]
Thanks. All I can do for now is add some debug logging, according to your own variables it should work as expected. In your console, you should already see the URL where you are being redirected to.

**Reply by sfuantos** [reply]
I just noticed that it only goes to /transactions instead of /firefly/transactions when I mass edit selecting a row and clicking the top Actions button (where the text changes to Edit selected individually or edit selected in bulk) instead of the Actions button in the row. Same result with edit selected individually or edit selected in bulk.

I just updated to 6.0.20.

**Reply by JC5** [reply]
Like I said, I added some debug information. Could you go to the development console in your browser (ctrl-shift-i) and tell me what it says the Mass Edit URL is?

**Reply by sfuantos** [reply]
The value of edit individually:
Mass edit URL is https://mydomain.com/transactions/mass/edit/623,622

Bulk:
Mass edit URL is https://mydomain.com/transactions/bulk/edit/623,622

**Reply by sfuantos** [reply]
Not sure if it´s related but something similar happens with categories when doing a search and then clicking in a category.
It goes to https://mydomain/categories/show/11 instead of https://mydomain/firefly/categories/show/11.

It resolves correctly when doing the same but on Transactions > Expenses.

**Reply by JC5** [reply]
Thanks, that helps. Could you pull the `develop` tag and see if that fixes it? You may have to clear your cache. In your browser console, I would expect something like this:

```
baseHref for mass edit is "https://firefly.dev.local/".
Bulk edit URL is https://firefly.dev.local//transactions/bulk/edit/118,117
```

**Reply by sfuantos** [reply]
on the develop tag seems to be working as expected in transactions
![Screenshot 2023-08-23 152929](https://github.com/firefly-iii/firefly-iii/assets/22817298/1444c523-1d59-46fb-95a4-8d2656aa0a9e)
![Screenshot 2023-08-23 161804](https://github.com/firefly-iii/firefly-iii/assets/22817298/1d9a0030-7aad-43ca-bf38-449e8a20936e)

but not working with categories
![Screenshot 2023-08-23 162053](https://github.com/firefly-iii/firefly-iii/assets/22817298/2fed510e-c26b-4296-989b-723ad8779ca5)

on 6.0.20 I get the Mass edit URL message and an error
![Screenshot 2023-08-23 152041](https://github.com/firefly-iii/firefly-iii/assets/22817298/6a41bbfc-456e-44d8-b1b9-66be1a2a3256)

In these last tests I was browsing as guest in Edge, so cache or cookies shouldn't be an issue.

**Reply by JC5** [reply]
Thanks, that matches my expectations. As for the categories from the search: probably because the results are loaded using AJAX, and your reverse proxy script doesn't edit them. I'll see what I can do!

**Reply by JC5** [reply]
Try the latest `develop`, I'm curious if this works as expected.

**Reply by KiralyCraft** [reply]
This also seems to be the issue for the image at the dashboard, namely "maskable144.png". The instance is hosted at /firefly/, so the image should be requested from /firefly/maskable144.png ; It is instead requested from /maskable144.png

**Reply by JC5** [reply]
Those images are referred to from the manifest.webmanifest file and the paths are already relative.

**Reply by KiralyCraft** [reply]
Something might be wrong then, with my configuration. I will investigate some more, as 99% of the website seems to work. Should I hit any roadblock, given that my setup is similar (with a subdirectory), should I open a new discussion, or open a new comment here and followup in the replies?

**Reply by JC5** [reply]
I suspect that your reverse proxy doesn't edit the paths, although I would expect the browser to respect the base href.

**Comment by paulius-valiunas** [comment]
https://github.com/orgs/firefly-iii/discussions/5904
